utility/types/message.go

```diff
@@ -0,0 +1,434 @@
+package types
+
+import (
+	"bytes"
+	"log"
+	"net/url"
+	"strconv"
+	"strings"
+
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/types"
+	"google.golang.org/protobuf/proto"
+)
+
+type Message interface {
+	proto.Message
+
+	SetSigner(signer []byte)
+	ValidateBasic() types.Error
+}
+
+var _ Message = &MessageStakeApp{}
+
+func (msg *MessageStakeApp) ValidateBasic() types.Error {
```

Comment:
Foor for thought:

```suggestion
  err := RunValidations(
    []Validations{
        "Amount": ValidateAmount,
        "PublicKey": ValidatePublicKey, 
        "Chains": ValidateRelayChains,
        "OutputAddress": ValidateOutputAddress,
    }
   )

   if err != nil {
       return false
   }

   return true
```

Reply:
Would shorten a lot of the existing functions, as you can imagine, and reduce the space for error.